### PR TITLE
Generate schedule from transcripts

### PR DESCRIPTION
**Pull request by thisRaptori**
Noticed that a bunch of things in the `schedule` weren't in the `transcripts`, and most of the ones which were didn't have a link.

Instead of updating the `schedule` manually, it's a lot easier to just add that extra data into the markdown frontmatter for the `transcript`, and generate the schedule from there!

Whenever we have an upcoming event, we just need to create a blank markdown file in the `transcripts` directory, add the title/date/other frontmatter, and it'll automatically show it in the `upcoming` section of the `schedule` page! 🎉

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -1,17 +1,54 @@
...
+];
+
+exports.onCreateNode = ({ node }) => {
+  if (node.frontmatter) {
+    frontmatterFieldSettings.forEach(
+      ({ field, defaultValue, supportMarkdown }) => {
+        const value = node.frontmatter[field];
+
+        if (supportMarkdown && value) {
+          node.frontmatter[field] = remark()
```
Comment (vcarl) [inline gatsby-node.js:20]: Interesting, so this is to support markdown in text values of frontmatter? But stripping paragraph tags?

File: src/pages/schedule.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import { graphql, Link } from 'gatsby';
+import Helmet from 'react-helmet';
+import MarkdownStyles from '../css/markdown-styles';
+import { Container, SmallTitle, MarkdownContainer } from '../utils/components';
+import Layout from '../utils/components/Layout';
+
+const currentDate = new Date();
+
+export default function Schedule({ data }) {
+  const { children } = data.transcripts.nodes.reduce(
```
Comment (vcarl) [inline src/pages/schedule.js]: This reduce reads pretty confusing to me. I'm gonna take a stab at a refactor—with the time comparison, flags, and accumulator values not used externally, I'm finding it hard to follow the logic. Might try and split it into separate "upcoming/current/past" events with a `filter()`, see how that looks. If it's not a clear win I'll open a PR to review separately (or just scrap it!).

Comment (vcarl) [reply src/pages/schedule.js]: Got something I feel very good about, so I pushed it directly. [Commit](https://github.com/reactiflux/reactiflux.com/pull/96/commits/17c1958562d937dc812c085499e6dad47acf11d3). Stripped it down to a couple helper functions, a component, and a pair of filters.

File: src/pages/schedule.js [hunk 3]
```diff
@@ -4,90 +4,29 @@ import Helmet from 'react-helmet';
...
-            time,
-            title,
-          },
-        },
-        name,
-        relativeDirectory,
-      } = node;
-
-      const isUpcoming = currentDate < new Date(date);
+  const currentDate = new Date();
```
Comment (vcarl) [inline src/pages/schedule.js:8]: Moved this on the off chance somebody leaves the site open for multiple days across when a Q&A happens. I think that would have rendered incorrectly until refresh because the date would only be evaluated at initial load, vs on page render with this change.

I suppose that this is gonna cause a flash when SSR output at build doesn't match client render as date drifts away from the last server build, but since we'll have to build to add new events that feels acceptable.

File: src/utils/groupBy.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export const groupBy = function(key, xs) {
+  console.log(xs);
```
Comment (vcarl) [inline src/utils/groupBy.js]: Oops
```suggestion
```

File: gatsby-node.js (hunk 1 above)
Comment (thisRaptori) [reply gatsby-node.js:20]: Yep! Kinda sucks to need to strip the paragraph tags, but it's needed cause I want to use this directly inside an `<li>` without messing up the formatting...

File: src/pages/schedule.js (hunk 2 above)
Comment (thisRaptori) [reply src/pages/schedule.js]: Cool, definitely an improvement! 😁

Wonder if it'd be a good idea to refactor these bits slightly:

https://github.com/reactiflux/reactiflux.com/blob/17c1958562d937dc812c085499e6dad47acf11d3/src/pages/schedule.js#L15-L20

https://github.com/reactiflux/reactiflux.com/blob/17c1958562d937dc812c085499e6dad47acf11d3/src/pages/schedule.js#L24-L29

into something like:

```js
const [upcomingEventNodes, pastEventNodes] = partition(
	(node) =>
          currentDate <= new Date(node.childMarkdownRemark.frontmatter.date),
	nodes.map(simplifyNode)
)

const upcomingEvents = groupBy('dateGroup', upcomingEventNodes)
const pastEvents = groupBy('dateGroup', pastEventNodes)
```

🤔

Comment (vcarl) [reply src/pages/schedule.js]: Ooo wait partition sounds awesome, I haven't used that. Yeah I like that

**Review by vcarl — APPROVED** [review]
Everything looks good to me!
